tests/analysis/test_location_identification.py

```diff
@@ -230,3 +231,125 @@ def test_freq_method(self, example_freq):
         li = location_identifier(example_freq, method="FREQ", pre_filter=False)
         fr = freq_method(example_freq)
         assert_geodataframe_equal(li, fr)
+
+    def test_osna_method(self, example_osna):
+        """Test if function calls osna method correctly."""
+        li = location_identifier(example_osna, method="OSNA", pre_filter=False)
+        osna = osna_method(example_osna)
+        assert_geodataframe_equal(li, osna)
+
+
+@pytest.fixture
+def example_osna():
+    """Example staypoints with 2 location for 2 users with 4 different times."""
+    weekday = "2021-05-19 "
+    weekend = "2021-05-22 "
+    t_rest = pd.Timestamp(weekday + "07:00:00", tz="utc")
+    t_work = pd.Timestamp(weekday + "18:00:00", tz="utc")
+    t_weekend = pd.Timestamp(weekend + "00:00:00", tz="utc")
+    t_leisure = pd.Timestamp(weekday + "01:00:00", tz="utc")
+    h = pd.Timedelta("1h")
+    list_dict = [
+        {"user_id": 0, "location_id": 0, "started_at": t_rest},
+        {"user_id": 0, "location_id": 0, "started_at": t_leisure},
+        {"user_id": 0, "location_id": 0, "started_at": t_work},  # (0, 0) 1 rest + 1 leisure + 1 work
+        {"user_id": 0, "location_id": 1, "started_at": t_rest},
+        {"user_id": 0, "location_id": 1, "started_at": t_work},
+        {"user_id": 0, "location_id": 1, "started_at": t_work},  # (0, 1) 1 rest + 2 work + 1 weekend
+        {"user_id": 0, "location_id": 1, "started_at": t_weekend},
+        {"user_id": 0, "location_id": 2, "started_at": t_leisure},  # (0, 2) 1 leisure
+        {"user_id": 1, "location_id": 0, "started_at": t_rest},
+        {"user_id": 1, "location_id": 0, "started_at": t_leisure},
+        {"user_id": 1, "location_id": 0, "started_at": t_work},  # (1, 0) 1 rest + 1 leisure + 1 work
+        {"user_id": 1, "location_id": 1, "started_at": t_leisure},
+        {"user_id": 1, "location_id": 1, "started_at": t_leisure},
+        {"user_id": 1, "location_id": 1, "started_at": t_work},
+        {"user_id": 1, "location_id": 1, "started_at": t_work},  # (1, 1) 2 leisure + 2 work
+        {"user_id": 1, "location_id": 2, "started_at": t_leisure},  # (1, 2) 1 leisure
+    ]  # I am not happy with this fixture as we test for a lot of things at the same time.
```

Comment:
really big example that is almost hard to grasp what is going on here. For me it is hard to understand what the tests are doing, perhaps you can try to split this into several fixtures.